### PR TITLE
SendEmail Workflow activity, workflow type casting, some security checks 

### PR DESCRIPTION
**Pull request by majakubowski**
- security checks for lookups in CreateRequest - principal must be able to Read and AppendTo on referenced record
- security check for RetrieveMultipleRequest - at least Basic permission in required to success
- added RetrieveAttributeRequest Handler
- enhancing ConvertType & GetEntityProperty to handle EntityReference/OptionSetValue/string to String conversion
- added SendEmail Workflow Activity

### CODE REVIEW

File: src/XrmMockupShared/Requests/CreateRequestHandler.cs [hunk 1]
```diff
@@ -28,9 +28,29 @@ internal class CreateRequestHandler : RequestHandler {
...
+                        $", but the calling user with id '{userRef.Id}' does not have create access for that entity");
+                }
+
+                var references = clonedEntity.Attributes.Where(x => x.Value is EntityReference && x.Key != "ownerid").ToArray();
+                foreach(var attr in references)
+                {
+                    var reference = attr.Value as EntityReference;
+                    if (!security.HasPermission(reference, AccessRights.ReadAccess, userRef))
+                    {
+                        throw new FaultException($"Trying to create entity '{entity.LogicalName}'" +
```
[the remark this answers is not in the document]
Comment (majakubowski) [reply src/XrmMockupShared/Requests/CreateRequestHandler.cs]: It's strange.. Few minutes ago I tested this on CRM 8.2 onprem and CRM 9.0 online - the behaviour is the same for some OOB entities - I use Lead.ParentContactId with some custom prepared role as testcase. But the second testcase is two **custom** (maybe this make difrence) entities - first with ownership type = User or Team.. and second with ownership type = **Organization** (maybe this make diffrence). On CRM 9.0 the behavoiur is the same like for first testcase, but on CRM 8.2 behaviour is diffrent - system require prvRead & prvAppendTo privilage for referenced entity. 

I will try address these findings & differences in code ;)

**Comment by majakubowski** [comment]
I also founded some things for fix in XrmMockupWorkflow project... I'm working on it - please wait with merging for my next commit.

File: src/XrmMockupShared/Requests/CreateRequestHandler.cs (hunk 1 above)
[the remark this answers is not in the document]
Comment (majakubowski) [reply src/XrmMockupShared/Requests/CreateRequestHandler.cs]: I moved ReadAccess check to UI mode.

**Comment by majakubowski** [comment]
resolves #69

**Comment by magesoe** [comment]
Features from this PR was added in 0.9.11 https://www.nuget.org/packages/XrmMockup365/0.9.11-beta

**Comment by majakubowski** [comment]
Thank you very much :)

**Pull request closed without merging**